### PR TITLE
Pull Req for controller tests for videos, authentication and reviews

### PR DESCRIPTION
**Pull request by shachopin**
Please review my pull request

### CODE REVIEW

File: spec/controllers/videos_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      expect(assigns(:video)).to eq(video)
+    end
+
+    it "sets @reviews for authenticated users" do
+      session[:id] = Fabricate(:user).id
+      video = Fabricate(:video)
+      review1 = Fabricate(:review, video: video)
+      review2 = Fabricate(:review, video: video)
+      get :show, id: video.id
+      assigns(:reviews).should =~ [review1, review2]
```
Comment (tomtomecek) [inline spec/controllers/videos_controller_spec.rb]: You are trying 2 different RSpec syntax styles, this is `should` syntax and elsewhere `expect` syntax style. But proper way is to stick and use 1 way, if you want to match then `match_array` matcher should be used.

File: logfile [hunk 2]
```diff
@@ -192,3 +192,53 @@ STATEMENT:  CREATE DATABASE "daweidai_development" ENCODING = 'unicode'
 ERROR:  database "daweidai_development" is being accessed by other users
 DETAIL:  There is 1 other session using the database.
 STATEMENT:  DROP DATABASE IF EXISTS "daweidai_development"
+LOG:  database system was shut down at 2015-10-13 09:11:44 PDT
```
Comment (tomtomecek) [inline logfile]: This file should be gitignored.

File: app/views/videos/show.html.haml [hunk 3]
```diff
@@ -12,3 +12,35 @@
...
+%section.reviews.container
+  .row
+    .col-sm-10.col-sm-offset-1
+      = form_for [@video, Review.new] do |f|
+        %fieldset
+          .form-group
+            %label Rate this video
+            .row
+              .col-sm-3
+                =f.select :rating, options_for_select([5,4,3,2,1].map{|number| [pluralize(number, "Star")]}), class:"form-control"
```
Comment (tomtomecek) [inline app/views/videos/show.html.haml]: Its better when you turn `options_for_select` to a helper method. too much logic in this template.

File: app/controllers/reviews_controller.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+class ReviewsController < ApplicationController
+  before_filter :require_login
```
Comment (tomtomecek) [inline app/controllers/reviews_controller.rb]: `before_action` is new way as `filter` will be deprecated in Rails 5.

File: logfile (hunk 2 above)
Comment (shachopin) [inline logfile]: I already have this line in my gitignore file

...
/log/*.log
logfile
/tmp

Do you know why it is still there?

**Comment by shachopin** [comment]
I just submitted another commit for this pull request. Hopefully it fixes all issues mentioned by you. But I don't know how to exclude the logifle. I already put that in gitignore. Not sure why it's still there. 

By the way, do you actually see the pull request being updated automatically whenever I make more commits?

Thank you for your help?

File: logfile (hunk 2 above)
Comment (tomtomecek) [inline logfile]: You have committed it to your git local repository. So something like `git rm logfile` - recommit and push should get rid of that.

**Comment by tomtomecek** [comment]
I get notified when you comment or merge the PR.

**Comment by tomtomecek** [comment]
Anyway it looks good, I checked the changes.

**Pull request merged**